### PR TITLE
Issue/202

### PR DESCRIPTION
**Pull request by VictorGTdaCruz**
#### What this PR does?
Update the spinner so that it can have a custom layout

#### How should it be manually tested?
Run sample activity

### CODE REVIEW

File: app/src/main/java/br/com/stone/emeraldcomponentsandroid/activities/SpinnerActivity.kt [hunk 1]
```diff
@@ -5,15 +5,25 @@ import android.util.Log
...
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_spinner)
+
         emeraldSpinner.setItems(listOf("Option 1", "Option 2"))
         emeraldSpinnerWithPlaceholder.setItems(listOf("Option 1", "Option 2"))
 
+        val customList = listOf("teste" to "subteste", "aaaa" to "bbbb", "32131" to "ffwef")
+        emeraldSpinnerWithCustomLayout.setItems(customList, R.layout.spinner_custom_layout) { position ->
+            @Suppress("UNCHECKED_CAST")
```
Comment (renanrbs) [inline app/src/main/java/br/com/stone/emeraldcomponentsandroid/activities/SpinnerActivity.kt:21]: Have you tried working with generics?

Comment (VictorGTdaCruz) [reply app/src/main/java/br/com/stone/emeraldcomponentsandroid/activities/SpinnerActivity.kt:21]: nice

File: app/src/main/java/br/com/stone/emeraldcomponentsandroid/activities/SpinnerActivity.kt [hunk 2]
```diff
@@ -5,15 +5,25 @@ import android.util.Log
...
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_spinner)
+
         emeraldSpinner.setItems(listOf("Option 1", "Option 2"))
         emeraldSpinnerWithPlaceholder.setItems(listOf("Option 1", "Option 2"))
 
+        val customList = listOf("teste" to "subteste", "aaaa" to "bbbb", "32131" to "ffwef")
+        emeraldSpinnerWithCustomLayout.setItems(customList, R.layout.spinner_custom_layout) { item ->
+            @Suppress("UNCHECKED_CAST")
+            val selectedItem = item as? Pair<String, String>
```
Comment (renanrbs) [inline app/src/main/java/br/com/stone/emeraldcomponentsandroid/activities/SpinnerActivity.kt:22]: Is this cast still needed?

File: app/src/main/java/br/com/stone/emeraldcomponentsandroid/activities/SpinnerActivity.kt [hunk 3]
```diff
@@ -5,15 +5,23 @@ import android.util.Log
...
 class SpinnerActivity : AppCompatActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_spinner)
+
         emeraldSpinner.setItems(listOf("Option 1", "Option 2"))
         emeraldSpinnerWithPlaceholder.setItems(listOf("Option 1", "Option 2"))
 
+        val customList = listOf("teste" to "subteste", "aaaa" to "bbbb", "32131" to "ffwef")
```
Comment (renanrbs) [inline app/src/main/java/br/com/stone/emeraldcomponentsandroid/activities/SpinnerActivity.kt:19]:
```suggestion
        val customList = listOf("test" to "subtest", "aaaa" to "bbbb", "123123" to "abcdef")
```

**Pull request merged**